### PR TITLE
refactor(evaluations): remove the run pass rate that can never be non-null

### LINKED ISSUE #2001: Run-level eval pass rate is structurally always null
## Summary

`passRate()` in `frontend/ui/src/lib/eval/pass-rate.ts:47-50` divides `passedCount` by `passedCount + failedCount` and returns `null` when that denominator is zero. **The denominator is always zero.** Those two counts increment only on `status === "passed"` / `"failed"` (`:35-36`), and no current SDK emits either status — `case_status()` returns `"errored"` or `"not_scored"` and nothing else (`traceroot/eval/results.py:130-140`; `packages/traceroot/src/eval/results.ts:181-184`), deliberately, because a case with several scorers has no single headline verdict. The runs list computes and ships `passRate` anyway; nothing in the repository reads it.

This is not a missing feature. A run-level pass/fail was **intentionally dropped from the product** — see `frontend/ui/src/features/evaluations/views/evaluations-view.tsx:28-31` ("per-run pass/fail and 'main score' are intentionally not surfaced here") and `views/run-detail-view.tsx:430-432` ("no headline metrics, status filters, pass-rate summary"). The rollup is stranded from a design that changed around it.

## How to see it

Run any evaluation and read the runs list response: `passRate` is `null`, `passedCount` and `failedCount` are `0`, while `erroredCount` / `notScoredCount` carry real values. No SDK code path changes this.

## Root cause (confirmed)

- `pass-rate.ts:47-50`, denominator structurally zero. Nothing consumes the output: `features/evaluations/types.ts` does not even declare a `passRate` field on the run type, and the only `passRate` in `features/evaluations/hooks.ts:398` is an **unrelated, live, scorer-level** field.
- Two comments assert a relationship that does not exist. `pass-rate.ts:3` and `types.ts:149` both cite a design document **not present in the repository**, and `pass-rate.ts:5-6` justifies the denominator as "inherited from the SDK's own definition (`results.py`: `scored_count = passed + failed`)" — a misread. `scored_count` counts how many cases produced a score at all.

## What must NOT be deleted — most of this module is live

1. **`countResultStatuses` stays.** Two of its four counts are live: `erroredCount` and `notScoredCount` are exactly the statuses `case_status()` returns. It has **two** call sites — `app/api/projects/[projectId]/evaluations/runs/route.ts:301` and `runs/[runId]/route.ts:145` — and both need it; tests assert non-zero values at `runs/route.test.ts:263-264` and `runs/[runId]/route.test.ts:251-252`.
2. **`excludedSummary` stays.** It reads only the two live counts and renders e.g. `"2 errored, 1 not scored"` (`runs/route.test.ts:307`, `:346`).
3. **Two different functions are named `passRate`.** `lib/eval/scorer-registry.ts:412` is live — it aggregates score-level `s.passed` as `passedTrue / passedTotal` and is the aggregate behind the scorer catalog. A grep-driven cleanup would delete working code. Only `lib/eval/pass-rate.ts`'s is dead.
4. **`"passed"` / `"failed"` stay in the status unions.** They look unreachable, but the union is **inbound public-API validation**: `backend/rest/schemas/eval.py:132` feeds `status: EvalResultStatus` on the request model at `:330`, and `frontend/packages/core/src/eval-contract.ts:74` feeds the Zod request schema at `:316`. No shipped release has emitted them — the eval SDK first shipped in py `0.2.0` / ts `0.3.0`, both already returning only `errored`/`not_scored` — but narrowing an accepted-value set on an inbound public contract is a breaking change for any client that sends one, and buys nothing: an unrolled-up status is simply ignored. The parity fixtures carrying `"status": "passed"` pin exactly this wire compatibility.
5. **Per-score `passed` and `@scorer(threshold=...)` stay** — live, documented, persisted to a real column, and `threshold` feeds the LLM-judge auto-version hash, so removing it would re-version every declaratively-versioned judge.

## Impact

Low user impact, high maintenance cost. Two always-zero fields ship on every run in two API responses, a function that can only return `null` is computed per request, and the comments explaining all of it point at a missing document and a wrong SDK relationship. The next reader either builds on the false premise or deletes the wrong `passRate`.

## Acceptance criteria

- `passRate()` removed from `pass-rate.ts`, along with the `passRate:` field on the runs list response (`runs/route.ts:325`) and its tests.
- `passedCount` / `failedCount` removed from `ResultStatusCounts`, from the fold in `countResultStatuses`, from **both** route responses, from `types.ts:151-152`, and from the tests asserting them. **This is a response-shape change** on the runs list and run detail endpoints — two fields disappear from every run object; no client reads them, but the removal should be reviewed as an API change, not a refactor.
- `countResultStatuses`, `erroredCount`, `notScoredCount`, `excludedSummary` and all of `lib/eval/scorer-registry.ts` are **unchanged**, with their existing tests passing untouched.
- The stale document references at `pass-rate.ts:3` and `types.ts:149` and the false denominator rationale at `pass-rate.ts:5-6` are deleted, not reworded.
- The `"passed"`/`"failed"` union members are retained, with a comment at `schemas/eval.py:132` and `eval-contract.ts:74` recording why: they are accepted-but-never-produced, and narrowing an inbound accepted-value set is a breaking change that buys nothing. Parity fixtures unchanged.
- The phantom references at `features/evaluations/evaluations.smoke.test.tsx:58-59` are removed: they point at a `PassRate` component and a "Passed column renders a real fraction" test, **neither of which exists**. The run-list columns are Timestamp / Eval Name / Run Name / Dataset / Cost / Avg Cost / Duration / Avg Duration / Actions.
- **No data migration.** `frontend/packages/core/prisma/schema.prisma:655` stores `status` as a free-form `VarChar`; eval scores live in Postgres only, so no analytics store is involved.

Separate follow-ups, one per SDK repository rather than part of this change: a `pass_threshold` / `passThreshold` option is accepted and stored but never read, so it never reaches a request payload and this repository has no reference to it. Tracked for TypeScript in traceroot-ai/traceroot-ts#146.

### PR DESCRIPTION
**Pull request by dark-sorceror**
## Summary

Fixes: #2001

`passRate()` in `frontend/ui/src/lib/eval/pass-rate.ts` (now `result-status-counts.ts`) divided `passedCount` by `passedCount + failedCount` and returned `null` on a zero denominator. **The denominator was structurally zero.** Those two counts incremented only on `status === "passed"` / `"failed"`, and no current SDK emits either — `case_status()` returns `errored` or `not_scored` and nothing else, deliberately: a case with several scorers has no single headline verdict, so the SDK records a per-score `passed` and derives no case-level pass/fail. The runs list computed and shipped `passRate` on every run anyway, and nothing in the repository read it — `features/evaluations/types.ts` never even declared the field on the run type.

This was not an unfinished feature. A run-level pass/fail was intentionally dropped from the product, and the code says so in two places: `views/evaluations-view.tsx:29-31` ("per-run pass/fail and 'main score' are intentionally not surfaced here") and `views/run-detail-view.tsx:431` ("no headline metrics, status filters, pass-rate summary"). The rollup was stranded from a design that changed around it.

## Removed

- `passRate()`, and the `passRate:` field it fed on the runs list response (`evaluations/runs/route.ts`).
- `passedCount` / `failedCount` from `ResultStatusCounts`, from the fold in `countResultStatuses`, from both route responses, and from `RunRow` in `types.ts`.
- A reference in `pass-rate.ts` and `types.ts` to a design document that is not in the repository.
- The module name itself: `pass-rate.ts` → `result-status-counts.ts`, since it no longer computes a rate. The import path `@/lib/eval/pass-rate` read as if one were still served.
- A false rationale: the denominator was justified as "inherited from the SDK's own definition (`results.py`: `scored_count = passed + failed`)". `scored_count` counts how many cases produced a score at all; it was never a pass/fail denominator.

## Kept, deliberately

Most of this module is live, and two distinct things are named `passRate`.

- **`countResultStatuses` stays**, with its two surviving counts (`erroredCount` / `notScoredCount` — exactly the statuses `case_status()` returns) and both call sites: `evaluations/runs/route.ts:301` and `evaluations/runs/[runId]/route.ts:145`.
- **`excludedSummary` stays.** It reads only the surviving counts and renders e.g. `"2 errored, 1 not scored"`.
- To be precise about what "surviving" means: those counts are derived and served correctly, but no view renders them today, and `excludedSummary` is not even declared on `RunRow` — both pre-existing. This change removes the one field that could *never* carry a value and leaves the ones that can.
- **`lib/eval/scorer-registry.ts` is untouched.** It exports a *different* `passRate` (`:412`) that aggregates score-level `s.passed` as `passedTrue / passedTotal` and backs the scorer catalog. It is live. A grep-driven cleanup would have deleted working code.
- **The `"passed"` / `"failed"` members of the status unions stay.** They look unreachable, but the union is **inbound public-API validation**: `backend/rest/schemas/eval.py:137` feeds `status: EvalResultStatus` on the result-upload request model, and `frontend/packages/core/src/eval-contract.ts:79` feeds the matching Zod request schema. No shipped release has ever emitted them — the eval SDK first shipped in py `0.2.0` / ts `0.3.0`, both already metric-first — but narrowing an accepted-value set on an inbound public contract is a breaking change for any client that sends one, and buys nothing: an unrolled-up status is simply ignored. Both contracts now carry a comment saying so, and a new test pins that the fold *ignores* those statuses rather than rolling them up, so a future reader does not restore a rollup that never had a denominator.

## Response-shape change

Two fields (`passedCount`, `failedCount`) disappear from every run object on both the runs list and the run detail endpoint, plus `passRate` from the list. Every one of them was always `0` or `null`, and no client in the repository reads them — but this should be reviewed as an API change, not a pure refactor.

**No data migration.** `status` is stored as a free-form `VarChar` and every row already holds `errored` or `not_scored`; eval data lives in Postgres only, so no analytics store is involved.

## Verification

- `frontend/ui` vitest: 210 files, 1966 tests passing. Re-adding `passedCount` to the fold fails 5 tests — four in the unit file and, deliberately, one at the route, so the guard does not rest on a single assertion.
- `frontend/packages/core` vitest: 12 files, 129 tests passing.
- Backend: `1558 passed, 5 skipped` — the cross-language contract-parity fixtures, which carry `"status": "passed"`, are unchanged and green.
- `pnpm lint`: 0 errors (pre-existing warnings only). Prettier clean on every touched file. `tsc --noEmit`: 6 pre-existing errors, none in a touched file.

## Out of scope

A `pass_threshold` field is accepted and stored but never read. It lives in the SDK repositories, not this one, so it is tracked separately.

### CODE REVIEW

File: frontend/packages/core/src/eval-contract.ts [hunk 1]
```diff
@@ -71,6 +71,12 @@ export const EVAL_RUN_STATUSES = [
 export const EvalRunStatusSchema = z.enum(EVAL_RUN_STATUSES);
 export type EvalRunStatus = (typeof EVAL_RUN_STATUSES)[number];
 
+// `passed`/`failed` are accepted but never produced: `caseStatus()` returns only
+// `errored`/`not_scored`, and no shipped release has emitted the older pair (the eval
+// SDK first shipped in 0.3.0, already metric-first). They stay because this enum backs
```
[the remark this answers is not in the document]
Comment (dark-sorceror) [reply frontend/packages/core/src/eval-contract.ts]: Resolved in ae846f4

**Review by XinweiHe — APPROVED** [review]
